src/util.c

```diff
@@ -45,3 +46,40 @@ void *ecalloc(size_t nmemb, size_t size)
         err(EXIT_FAILURE, "calloc");
     return p;
 }
+
+void *erealloc(void *ptr, size_t size)
+{
+    void *p = realloc(ptr, size);
+    if (!p)
+        err(EXIT_FAILURE, "realloc");
+    return p;
+}
+
+void streamReserve(Stream *buf, size_t size)
+{
+    assert(buf->off <= buf->cap);
```

Comment:
I don't like `assert()`. Either the check is worth putting in the release build or it's not worth having at all.

Reply:
> Either the check is worth putting in the release build or it's not worth having at all.

Different tools for different purposes. Things like address sanitizer or even the `-g` flag of compiler are examples of tools that are meant to be used for debugging purposes. It doesn't make sense to say, "I won't use `-g` or sanitizers because it's not enabled in release build."

Assertions are no different in this regard, it's a debugging tool for debugging purposes - to catch a bug as early as possible instead of marching forward as if everything is OK.

- - -

On a sidenote, this also makes me wonder if we actually ever define `NDEBUG` or not. Probably makes sense to have it defined in the release tarballs (assuming it's not done already).